tensorflow/compiler/tf2tensorrt/convert/utils.h

```diff
@@ -128,77 +129,199 @@ inline bool IsTrtShapeTensorCompatible(const Tensor& tensor) {
+    result->nbDims = num_dims_;
+    absl::c_copy(storage_, static_cast<int32_t*>(result->d));
     return Status::OK();
   }
-  return ContainerToTrtDims(shape.dim_sizes(), trt_dims, ignore_first_dim);
-}
 
-Status GetNetworkInputShapes(const nvinfer1::INetworkDefinition* network,
-                             std::vector<PartialTensorShape>* input_shapes);
+  // Convert to an nvinfer1::Dims and return by value.
+  StatusOr<nvinfer1::Dims> AsTrtDims() const {
+    nvinfer1::Dims result;
+    TF_RETURN_IF_ERROR(TrtDims(&result));
+    return result;
+  }
 
-// Creates PartialTensorShape from nvinfer1::Dims. The "batch_dimension", when
-// provided, is prepended to the shape.
-inline PartialTensorShape TrtDimsToPartialTensorShape(
-    const nvinfer1::Dims& dims,
-    absl::optional<int64> batch_dimension = absl::nullopt) {
-  absl::InlinedVector<int64, 4> dims_vec;
-  if (batch_dimension) {
-    dims_vec.insert(dims_vec.begin(), std::max(int64(-1), *batch_dimension));
-  }
-  std::copy(dims.d, dims.d + dims.nbDims, std::back_inserter(dims_vec));
-  return PartialTensorShape(dims_vec);
-}
+  // Convert to a TensorShape.
+  Status TensorShape(TensorShape* shape,
+                     absl::optional<int> batch_size = absl::nullopt) const {
+    TF_RETURN_IF_ERROR(status_);
+    TF_RETURN_IF_ERROR(TensorShapeUtils::MakeShape(
+        static_cast<const int64_t*>(storage_.data()), storage_.size(), shape));
+    if (batch_size) shape->InsertDim(0, *batch_size);
+    return Status::OK();
+  }
 
-Status TrtDimsToTensorShape(const std::vector<int>& trt_dims,
-                            TensorShape* shape,
-                            absl::optional<int> batch_size = absl::nullopt);
+  Status PartialTensorShape(
+      PartialTensorShape* shape,
+      absl::optional<int> batch_size = absl::nullopt) const {
+    TF_RETURN_IF_ERROR(status_);
+    TF_RETURN_IF_ERROR(TensorShapeUtils::MakeShape(
+        static_cast<const int64_t*>(storage_.data()), storage_.size(), shape));
+    if (batch_size) shape->InsertDim(0, *batch_size);
+    return Status::OK();
+  }
 
-template <typename TensorShapeType>
-Status TrtDimsToTensorShape(const nvinfer1::Dims trt_dims,
-                            TensorShapeType* shape,
-                            absl::optional<int> batch_size = absl::nullopt) {
-  TF_RETURN_IF_ERROR(
-      TensorShapeUtils::MakeShape(trt_dims.d, trt_dims.nbDims, shape));
-  if (batch_size) {
-    shape->InsertDim(0, batch_size.value());
-  }
-  return Status::OK();
-}
+  Status Vector(std::vector<int32_t>* shape) const {
+    TF_RETURN_IF_ERROR(status_);
+    shape->clear();
+    absl::c_copy(storage_, std::back_inserter(*shape));
+    return Status::OK();
+  }
+
+  //----- Property Accessors ------
+
+  // Returns true if the shape should be interpreted as a scalar. This follows
+  // TensorRT conversions: a scalar shape can have NumDims()==1 or NumDims()==0,
+  // but the underlying storage_ container has a single dimension of size 1.
```

Comment:
This comment doesn't seem to match the implementation, does it?
Maybe you want something like this?
 Returns true if the dimension values represent a static shape.

Why we don't need to check status_ field here?

Reply:
Yes, wrong comment... see above regarding checking status_.

Reply:
Will update to StatusOr<bool>